tests/ion_schema_tests_runner.rs

```diff
@@ -0,0 +1,167 @@
+
+        for element in schema_content {
+            let annotations: Vec<&OwnedSymbolToken> = element.annotations().collect();
+            if annotations.contains(&&text_token("invalid_type")) {
+                // check for an invalid type validation
+                let type_def_result = schema_system.schema_type_from_element(&element, type_store);
+                if type_def_result.is_ok() {
+                    println!("ERROR: expected error for invalid type {:?}", element);
+                    test_failed = true;
+                    break;
+                }
+            } else if annotations.contains(&&text_token("invalid")) {
+                // get invalid values to validate
+                invalid_values = element
+                    .as_sequence()
+                    .unwrap()
+                    .iter()
+                    .map(|v| v.to_owned())
+                    .collect();
+            } else if annotations.contains(&&text_token("valid")) {
+                // get valid values to validate
+                valid_values = element
+                    .as_sequence()
+                    .unwrap()
+                    .iter()
+                    .map(|v| v.to_owned())
+                    .collect();
+            } else if annotations.contains(&&text_token("type")) {
+                // get type definition for type validation
+                type_def = Some(
+                    schema_system
+                        .schema_type_from_element(&element, type_store)
+                        .unwrap(),
+                );
+            } else {
+                continue;
+            }
+        }
+
+        match type_def {
+            None => {}
+            Some(schema_type) => {
+                for valid_value in valid_values {
+                    match schema_type.validate(&valid_value, type_store) {
+                        Ok(_) => {
+                            continue;
+                        }
+                        Err(error) => {
+                            println!("ERROR: {} for {:?}", error, valid_value);
+                            test_failed = true;
+                        }
+                    };
+                }
+
+                if test_failed {
+                    // move to next test file if any test failed
+                    continue;
+                }
+
+                for invalid_value in invalid_values {
+                    match schema_type.validate(&invalid_value, type_store) {
+                        Ok(_) => {
+                            println!(
+                                "ERROR: expected invalid value to return error for {:?}",
+                                &invalid_value
+                            );
+                            test_failed = true;
+                            break;
+                        }
+                        Err(_) => {
+                            continue;
+                        }
+                    };
```

Comment:
Ideally, all test failures should be reported, not just the first one encountered, so I don't think we should have the `break`. Also, I _think_ this can be simplified.
```suggestion
                    if let Ok(_) = schema_type.validate(&invalid_value, type_store) {
                        println!(
                            "ERROR: expected invalid value to return error for {:?}",
                            &invalid_value
                        );
                        test_failed = true;
                    };
```